docs/operations/extraction.md

```diff
@@ -0,0 +1,136 @@
+
+# extraction #
+
+This processor encapsulates the extraction phase of [ts-transforms](https://terascope.github.io/teraslice/docs/packages/ts-transforms/overview). This is meant to work in conjunction with the selection processor as this works by using the metadata set in the selection processor.
+
+
+
+
+| Configuration | Description | Type |  Notes |
+| --------- | -------- | ------ | ------ |
+| _op | Name of operation, it must reflect the exact name of the file | String | required |
+| rules | an array of strings that are the locations where rule files. must be specifed in "assetName:path" format | String[] | required |
+| plugins | an array of strings that are the locations where [plugins](https://terascope.github.io/teraslice/docs/packages/ts-transforms/plugins) reside. must be specifed in "assetName:modulePath" format | Object[] | optional, defaults to [] |
+| type_config | a schema for the data being consumed. Set the keys to your data's field names, with values set to this [enum](https://terascope.github.io/teraslice/docs/packages/types/api/enums/xlucenefieldtype) | Object | optional |
+| variables | An object containing any varialbes for the xlucene rules | Object | optional|
+
+
+## Usage
+
+```typescript
+
+// someAssetId:transformRules.txt
+{ "selector": "some: $foo", "source_field": "field", "target_field": "interm1", "tag": "someTag", "output": false }
+{ "selector": "some: $foo", "source_field": "field2", "target_field": "interm2", "tag": "someTag", "output": false }
+{ "follow": "someTag", "post_process": "join", "target_field": "final", "delimiter": " " }
+
+{ "source_field": "otherField", "target_field": "lastField", "other_match_required": true }
+{ "source_field": "_id", "target_field": "id", "other_match_required": true, "tag": "numberMe" }
+{ "follow": "numberMe", "post_process": "number" }
+
+{ "selector": "location: geoBox( top_left: '33.906320, -112.758421' bottom_right: '32.813646,-111.058902')", "source_field": "location", "target_field": "loc" }
+
+{ "selector": "date:[2019-04-16T20:14:44.304Z TO *] AND bytes:>=1000000", "source_field": "date", "target_field": "last_seen", "tag": "tagOp" }
+{ "follow": "tagOp", "post_process": "tag" }
+
+// ------------
```

Comment:
This should be a sperate code block

```suggestion

`someAssetId:transformRules.txt`
```json
{ "selector": "some: $foo", "source_field": "field", "target_field": "interm1", "tag": "someTag", "output": false }
{ "selector": "some: $foo", "source_field": "field2", "target_field": "interm2", "tag": "someTag", "output": false }
{ "follow": "someTag", "post_process": "join", "target_field": "final", "delimiter": " " }

{ "source_field": "otherField", "target_field": "lastField", "other_match_required": true }
{ "source_field": "_id", "target_field": "id", "other_match_required": true, "tag": "numberMe" }
{ "follow": "numberMe", "post_process": "number" }

{ "selector": "location: geoBox( top_left: '33.906320, -112.758421' bottom_right: '32.813646,-111.058902')", "source_field": "location", "target_field": "loc" }

{ "selector": "date:[2019-04-16T20:14:44.304Z TO *] AND bytes:>=1000000", "source_field": "date", "target_field": "last_seen", "tag": "tagOp" }
{ "follow": "tagOp", "post_process": "tag" }
```
```typescript
```